integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientLoanCreditBalanceRefundandRepaymentTypeIntegrationTest.java

```diff
@@ -438,4 +439,76 @@ public void newGoodwillCreditCreatesCorrectJournalEntriesForCashAccountingTest()
 
     }
 
+    @Test
+    public void tryToUndoGoodWillCreditTransactionTest() {
```

Comment:
Please amend this test cases to tests for testing whether:
- these transactions can be undone
- these transactions cannot be adjusted

Reply:
Done